recipes/cairo/all/conanfile.py

```diff
@@ -0,0 +1,258 @@
+            if self.options.shared:
+                self.copy(pattern="*cairo.lib", dst="lib", src=src, keep_path=False)
+                self.copy(pattern="*cairo.dll", dst="bin", src=src, keep_path=False)
+                self.copy(pattern="*cairo-gobject.lib", dst="lib", src=cairo_gobject, keep_path=False)
+                self.copy(pattern="*cairo-gobject.dll", dst="bin", src=cairo_gobject, keep_path=False)
+            else:
+                self.copy(pattern="*cairo-static.lib", dst="lib", src=src, keep_path=False)
+                self.copy(pattern="*cairo-gobject.lib", dst="lib", src=cairo_gobject, keep_path=False)
+                shutil.move(os.path.join(self.package_folder, 'lib', "cairo-static.lib"),
+                            os.path.join(self.package_folder, 'lib', "cairo.lib"))
+        else:
+            with tools.chdir(self._source_subfolder):
+                env_build = self._get_env_build()
+                env_build.install()
+        tools.remove_files_by_mask(self.package_folder, "*.la")
+        
+        self.copy("COPYING*", src=self._source_subfolder, dst="licenses", keep_path=False)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+
+    def package_info(self):
+        self.cpp_info.components["cairo_"].names["pkg_config"] = "cairo"
+        self.cpp_info.components["cairo_"].libs = ["cairo"]
+        self.cpp_info.components["cairo_"].includedirs = [os.path.join('include', 'cairo')]
+        self.cpp_info.components["cairo_"].requires = ["pixman::pixman", "libpng::libpng", "zlib::zlib"]
+        if self.options.with_freetype:
+            self.cpp_info.components["cairo_"].requires.append("freetype::freetype")
+        
+        if self.settings.os == "Windows":
+            if not self.options.shared:
+                self.cpp_info.components["cairo_"].defines.append('CAIRO_WIN32_STATIC_BUILD=1')
+        else:
+            if self.options.with_glib:
+                self.cpp_info.components["cairo_"].requires.extend(["glib::gobject-2.0", "glib::glib-2.0"])
+            if self.options.with_fontconfig:
+                self.cpp_info.components["cairo_"].requires.append("fontconfig::fontconfig")
+        if self.settings.os == "Linux":
+            if self.options.with_xcb:
+                self.cpp_info.components["cairo_"].requires.extend(["xorg::xcb-shm", "xorg::xcb"])
+            if self.options.with_xlib_xrender:
+                self.cpp_info.components["cairo_"].requires.extend(["xorg::xcb-render"])
+            if self.options.with_xlib:
+                self.cpp_info.components["cairo_"].requires.extend(["xorg::x11", "xorg::xext"])
+
+
+        if self.settings.os == "Windows":
+            self.cpp_info.components["cairo-win32"].names["pkg_config"] = "cairo-win32"
+            self.cpp_info.components["cairo-win32"].requires = ["cairo_", "pixman::pixman", "libpng::libpng"]
+            self.cpp_info.components["cairo-win32"].includedirs = [os.path.join('include', 'cairo')]
```

Comment:
yes

Reply:
Actually it is necessary to add this include directory, in order to generate usable .pc files, otherwise `pkg-config --cflags cairo-gobject` et al would not return the include path to cairo

Reply:
`cairo-gobject` has `cairo_` in its requirements lists.
And `cairo_` already sets the `includedirs`.
So I was wondering whether it is needed to do this twice.

Reply:
yes, it is needed, because pkg-config cflags are not transitive

Reply:
I just tested this on msys2 and this is incorrect.
Run the following:
```
cat >/tmp/foo.pc <<EOF
prefix=/usr/local
exec_prefix=\${prefix}
includedir=\${prefix}/include
libdir=\${exec_prefix}/lib

Name: foo
Description: The foo library
Version: 1.0.0
Cflags: -I\${includedir}/foo -DFOO
Libs: -L\${libdir} -lfoo
EOF

cat >/tmp/bar.pc <<EOF
prefix=/usr/local
exec_prefix=\${prefix}
includedir=\${prefix}/include
libdir=\${exec_prefix}/lib

Name: bar
Description: The bar library
Version: 1.0.0
Requires: foo
Cflags: -I\${includedir}/bar -DBAR
Libs: -L\${libdir} -lbar
EOF

PKG_CONFIG_PATH=/tmp pkg-config bar --cflags
```

The last command prints:
```
-DBAR -DFOO -I/usr/local/include/bar -I/usr/local/include/foo
```
which is a combination of both `bar.pc` and `foo.pc`.

Reply:
woopsie you're right. I said that because I tried `pkg-config --cflags cairo-gobject` using system provided pc files, and I got a single `-I/usr/include/cairo `, but the actual reason is that pkg-config does some de-duplication.